carapace-server/src/main/java/org/carapaceproxy/server/mapper/requestmatcher/HttpsRequestMatcher.java

```diff
@@ -31,8 +30,8 @@
 public class HttpsRequestMatcher implements RequestMatcher {
 
     @Override
-    public boolean matches(HttpRequest request) {
-        return RequestHandler.PROTO_HTTPS.equals(request.headers().get(RequestHandler.HEADER_X_FORWARDED_PROTO, ""));
+    public boolean matches(MatchingContext context) throws MatchingException {
+        return (boolean) context.getProperty(PROPERTY_HTTPS);
```

Comment:
we can have a shot circuit MatchingContext.isSecure() without breaking the abstraction